docs/source/Mapping.rst

```diff
@@ -0,0 +1,175 @@
+Quantum Circuit Mapping
+=======================
+
+Many quantum computing architectures limit the pairs of qubits that two-qubit operations can be applied to.
+This is commonly described by a device's *coupling map*.
+To execute a generic quantum circuit (with arbitrary interactions between its qubits) on such an architecture, the circuit needs to be *mapped*.
+This involves *qubit allocation*, where logical qubits are assigned to physical qubits in an *initial layout*, and *routing*, where the original circuit is augmented with SWAP gates such that it adheres to the target device's *coupling map*.
+
+Consider the following circuit.
+
+    .. code-block:: python3
+
+        from mqt import qmap
+        from qiskit import QuantumCircuit
+
+        qc = QuantumCircuit(4)
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.cx(0, 2)
+        qc.cx(0, 3)
+
+        qc.barrier()
+
+        qc.t(0)
+        qc.t(1)
+        qc.t(2)
+        qc.t(3)
+
+        qc.barrier()
+
+        qc.cx(0, 3)
+        qc.cx(0, 2)
+        qc.cx(0, 1)
+
+        qc.measure_all()
+
+        print(qc.draw(fold=-1))
+
+    .. code-block:: console
+
+               ┌───┐                ░ ┌───┐ ░                 ░ ┌─┐
+          q_0: ┤ H ├──■────■────■───░─┤ T ├─░───■────■────■───░─┤M├─────────
+               └───┘┌─┴─┐  │    │   ░ ├───┤ ░   │    │  ┌─┴─┐ ░ └╥┘┌─┐
+          q_1: ─────┤ X ├──┼────┼───░─┤ T ├─░───┼────┼──┤ X ├─░──╫─┤M├──────
+                    └───┘┌─┴─┐  │   ░ ├───┤ ░   │  ┌─┴─┐└───┘ ░  ║ └╥┘┌─┐
+          q_2: ──────────┤ X ├──┼───░─┤ T ├─░───┼──┤ X ├──────░──╫──╫─┤M├───
+                         └───┘┌─┴─┐ ░ ├───┤ ░ ┌─┴─┐└───┘      ░  ║  ║ └╥┘┌─┐
+          q_3: ───────────────┤ X ├─░─┤ T ├─░─┤ X ├───────────░──╫──╫──╫─┤M├
+                              └───┘ ░ └───┘ ░ └───┘           ░  ║  ║  ║ └╥┘
+       meas: 4/══════════════════════════════════════════════════╩══╩══╩══╩═
+                                                           0  1  2  3
+
+
+Now assume this circuit shall be mapped to a 4-qubit architecture defined by the following coupling map:
+
+.. image:: /images/linear_arch.svg
+   :width: 80%
+   :alt: 5-qubit Architecture
+   :align: center
+
+
```

Comment:
in the actual documentation the whitespace below the image kind of seems too small. I am not quite sure whether this can be changed easily. Feel free to ignore this. Just looks kind of weird.

Reply:
Also took notice of it. I'll try to find a quick fix

Reply:
Alright, so inserting a "|" (vertical bar) does the job